src/ripple/protocol/impl/STAmount.cpp

```diff
@@ -1261,6 +1261,8 @@ divRound (STAmount const& num, STAmount const& den,
 std::uint64_t
 mulDiv(std::uint64_t value, std::uint64_t mul, std::uint64_t div)
 {
+    if ((value == 0 || mul == 0) && div != 0)
+        return 0;
```

Comment:
I have no problem taking it out of `STAmount.cpp`, since it really is more of a basic utility function, and was only supposed to be wedged in there until we found the right place for it. I'm just not sure where the right place for it would be. I'm thinking one of:

* `beast/module/core/maths`
* `ripple/basics`

Any opinions.